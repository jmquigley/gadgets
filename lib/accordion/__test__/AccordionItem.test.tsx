'use strict';

import * as assert from 'assert';
import {mount, shallow} from 'enzyme';
import * as React from 'react';
import {AccordionItem, getDefaultAccordionItemProps} from '../index';

test('Test retrieval of AccordionItem props object', () => {
	const props = getDefaultAccordionItemProps();

	assert(props);
	expect(props).toMatchSnapshot();
});

test('Test the creation of a AccordionItem control', () => {
	const ctl = shallow(
		<AccordionItem className="test-class" title="Test Title" />
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test('Test the disabling of the AccordionItem', () => {
	const ctl = shallow(
		<AccordionItem
			disabled
			title="Test Title"
		/>
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test('Test making the AccordionItem invisible', () => {
	const ctl = shallow(
		<AccordionItem
			title="Test Title"
			visible={false}
		/>
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test('Test clicking of the AccordionItem header', () => {
	const click = jest.fn();
	const child = <div>child content</div>;
	const ctl = mount(
		<AccordionItem
			initialToggle={false}
			onClick={click}
			title="Test Title"
		>
			{child}
		</AccordionItem>
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();

	assert(!ctl.state('toggle'));
	assert(!ctl.contains(child));
	ctl.find('.ui-label').simulate('click');
	expect(click).toHaveBeenCalled();
	assert(ctl.state('toggle'));
	assert(ctl.contains(child));
});
