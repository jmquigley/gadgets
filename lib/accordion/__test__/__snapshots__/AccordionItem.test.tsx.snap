// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test clicking of the AccordionItem header 1`] = `
.c4 {
  background-color: inherit;
  font-size: 1rem;
}

.c2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: block;
  text-align: left;
  -webkit-flex: 5;
  -ms-flex: 5;
  flex: 5;
  font-size: 1rem;
}

.c1 {
  background-color: #ffffff;
  color: #000000;
  cursor: default;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1:hover {
  background-color: #c0c0c0 !important;
}

.c1:hover .ui-item-button {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  opacity: 1.0;
}

.c0 > .ui-item,
.c0 > .ui-item-button {
  color: #ffffff;
  background-color: #004358;
}

.c0 > .ui-item:hover {
  color: #ffe11a;
  background-color: #004358 !important;
}

<AccordionItem
  className=""
  contentEditable={false}
  direction="right"
  disabled={false}
  focus={false}
  height=""
  hiddenLeftButton={false}
  hiddenRightButton={false}
  id=""
  initialToggle={false}
  layout={5}
  leftButton={null}
  location="none"
  nocollapse={false}
  noedit={false}
  nohover={false}
  noripple={false}
  obj="AccordionItem"
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onDoubleClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onMouseOut={[Function]}
  onUpdate={[Function]}
  rightButton={null}
  selected={false}
  sizing="normal"
  stacked={false}
  style={Object {}}
  testing={true}
  theme={null}
  title="Test Title"
  tooltip=""
  useedit={false}
  visible={true}
  widget={null}
  width=""
  xcss={null}
>
  <ThemeProvider
    theme={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#c0c0c0",
        "color": "#000000",
        "headerBackgroundColor": "#004358",
        "headerForegroundColor": "#ffffff",
        "headerHoverColor": "#ffe11a",
        "hoverColor": "#c0c0c0",
        "inputBorderColor": "#c0c0c0",
        "selectedBackgroundColor": "#1f8a70",
        "selectedForegroundColor": "#ffe11a",
        "tooltipBackgroundColor": "#708090",
        "tooltipForegroundColor": "#ffffff",
        "transitionDelay": "0.5s",
      }
    }
  >
    <styled.ul
      className="ui-accordionitem"
      style={Object {}}
    >
      <ul
        className="ui-accordionitem c0"
        style={Object {}}
      >
        <Item
          className=""
          contentEditable={false}
          direction="right"
          disabled={false}
          focus={false}
          height=""
          hiddenLeftButton={false}
          hiddenRightButton={false}
          id=""
          initialToggle={false}
          layout={5}
          leftButton={null}
          location="none"
          nocollapse={false}
          noedit={false}
          nohover={false}
          noripple={false}
          obj="AccordionItem"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDoubleClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onMouseOut={[Function]}
          onUpdate={[Function]}
          rightButton={null}
          selected={false}
          sizing="normal"
          stacked={false}
          style={Object {}}
          testing={true}
          theme={null}
          title="Test Title"
          tooltip=""
          useedit={false}
          visible={true}
          widget={null}
          width=""
          xcss={null}
        >
          <ThemeProvider
            theme={
              Object {
                "backgroundColor": "#ffffff",
                "borderColor": "#c0c0c0",
                "color": "#000000",
                "headerBackgroundColor": "#004358",
                "headerForegroundColor": "#ffffff",
                "headerHoverColor": "#ffe11a",
                "hoverColor": "#c0c0c0",
                "inputBorderColor": "#c0c0c0",
                "selectedBackgroundColor": "#1f8a70",
                "selectedForegroundColor": "#ffe11a",
                "tooltipBackgroundColor": "#708090",
                "tooltipForegroundColor": "#ffffff",
                "transitionDelay": "0.5s",
              }
            }
          >
            <styled.li
              className="ui-item"
              id=""
              nohover={false}
              onBlur={[Function]}
              onDoubleClick={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onMouseOut={[Function]}
              selected={false}
              style={Object {}}
            >
              <li
                className="ui-item c1"
                id=""
                onBlur={[Function]}
                onDoubleClick={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onMouseOut={[Function]}
                selected={false}
                style={Object {}}
              >
                <Title
                  className=""
                  contentEditable={false}
                  direction="right"
                  disabled={false}
                  focus={false}
                  height=""
                  hiddenLeftButton={false}
                  hiddenRightButton={false}
                  id=""
                  initialToggle={false}
                  layout={5}
                  leftButton={null}
                  location="none"
                  nocollapse={false}
                  noedit={false}
                  nohover={false}
                  noripple={false}
                  obj="AccordionItem"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onDoubleClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onMouseOut={[Function]}
                  onUpdate={[Function]}
                  rightButton={null}
                  selected={false}
                  sizing="normal"
                  stacked={false}
                  style={
                    Object {
                      "padding": "2px 4px",
                    }
                  }
                  testing={true}
                  theme={null}
                  title="Test Title"
                  tooltip=""
                  useedit={false}
                  visible={true}
                  widget={null}
                  width=""
                  xcss={null}
                >
                  <ThemeProvider
                    theme={
                      Object {
                        "backgroundColor": "#ffffff",
                        "borderColor": "#c0c0c0",
                        "color": "#000000",
                        "headerBackgroundColor": "#004358",
                        "headerForegroundColor": "#ffffff",
                        "headerHoverColor": "#ffe11a",
                        "hoverColor": "#c0c0c0",
                        "inputBorderColor": "#c0c0c0",
                        "selectedBackgroundColor": "#1f8a70",
                        "selectedForegroundColor": "#ffe11a",
                        "tooltipBackgroundColor": "#708090",
                        "tooltipForegroundColor": "#ffffff",
                        "transitionDelay": "0.5s",
                      }
                    }
                  >
                    <styled.div
                      className="ui-title-bar ripple"
                      disabled={false}
                      layout={5}
                      style={
                        Object {
                          "padding": "2px 4px",
                        }
                      }
                      visible={true}
                    >
                      <div
                        className="ui-title-bar ripple c2"
                        disabled={false}
                        style={
                          Object {
                            "padding": "2px 4px",
                          }
                        }
                      >
                        <Styled(Label)
                          className="ui-title"
                          contentEditable={false}
                          direction="right"
                          disabled={false}
                          focus={false}
                          height=""
                          hiddenLeftButton={false}
                          hiddenRightButton={false}
                          id=""
                          initialToggle={false}
                          layout={5}
                          leftButton={null}
                          location="none"
                          nocollapse={false}
                          noedit={false}
                          nohover={false}
                          noripple={false}
                          obj="AccordionItem"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={[Function]}
                          onDoubleClick={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyPress={[Function]}
                          onMouseOut={[Function]}
                          onUpdate={[Function]}
                          rightButton={null}
                          selected={false}
                          sizing="normal"
                          stacked={false}
                          style={
                            Object {
                              "padding": "2px 4px",
                            }
                          }
                          testing={true}
                          text="Test Title"
                          theme={null}
                          title="Test Title"
                          tooltip=""
                          useedit={false}
                          visible={true}
                          widget={null}
                          width=""
                          xcss={
                            Array [
                              "
	text-align: left;
	flex: 5;
",
                            ]
                          }
                        >
                          <Label
                            className="ui-title c3"
                            contentEditable={false}
                            direction="right"
                            disabled={false}
                            focus={false}
                            height=""
                            hiddenLeftButton={false}
                            hiddenRightButton={false}
                            id=""
                            initialToggle={false}
                            layout={5}
                            leftButton={null}
                            location="none"
                            nocollapse={false}
                            noedit={false}
                            nohover={false}
                            noripple={false}
                            obj="AccordionItem"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onClick={[Function]}
                            onDoubleClick={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyPress={[Function]}
                            onMouseOut={[Function]}
                            onUpdate={[Function]}
                            rightButton={null}
                            selected={false}
                            sizing="normal"
                            stacked={false}
                            style={
                              Object {
                                "padding": "2px 4px",
                              }
                            }
                            testing={true}
                            text="Test Title"
                            theme={null}
                            title="Test Title"
                            tooltip=""
                            useedit={false}
                            visible={true}
                            widget={null}
                            width=""
                            xcss={
                              Array [
                                "
	text-align: left;
	flex: 5;
",
                              ]
                            }
                          >
                            <ThemeProvider
                              theme={
                                Object {
                                  "backgroundColor": "#ffffff",
                                  "borderColor": "#c0c0c0",
                                  "color": "#000000",
                                  "headerBackgroundColor": "#004358",
                                  "headerForegroundColor": "#ffffff",
                                  "headerHoverColor": "#ffe11a",
                                  "hoverColor": "#c0c0c0",
                                  "inputBorderColor": "#c0c0c0",
                                  "selectedBackgroundColor": "#1f8a70",
                                  "selectedForegroundColor": "#ffe11a",
                                  "tooltipBackgroundColor": "#708090",
                                  "tooltipForegroundColor": "#ffffff",
                                  "transitionDelay": "0.5s",
                                }
                              }
                            >
                              <styled.span
                                className="ui-label ui-title c3"
                                contentEditable={false}
                                disabled={false}
                                innerRef={[Function]}
                                onBlur={[Function]}
                                onClick={[Function]}
                                onDoubleClick={[Function]}
                                onKeyDown={[Function]}
                                onKeyPress={[Function]}
                                sizing="normal"
                                style={
                                  Object {
                                    "padding": "2px 4px",
                                  }
                                }
                                suppressContentEditableWarning={true}
                                visible={true}
                              >
                                <span
                                  className="ui-label ui-title c3 c4"
                                  contentEditable={false}
                                  disabled={false}
                                  onBlur={[Function]}
                                  onClick={[Function]}
                                  onDoubleClick={[Function]}
                                  onKeyDown={[Function]}
                                  onKeyPress={[Function]}
                                  style={
                                    Object {
                                      "padding": "2px 4px",
                                    }
                                  }
                                  suppressContentEditableWarning={true}
                                >
                                  Test Title
                                </span>
                              </styled.span>
                            </ThemeProvider>
                          </Label>
                        </Styled(Label)>
                      </div>
                    </styled.div>
                  </ThemeProvider>
                </Title>
              </li>
            </styled.li>
          </ThemeProvider>
        </Item>
      </ul>
    </styled.ul>
  </ThemeProvider>
</AccordionItem>
`;

exports[`Test clicking the AccordionItem header when setting nocollapse 1`] = `
.c4 {
  background-color: inherit;
  font-size: 1rem;
}

.c2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: block;
  text-align: left;
  -webkit-flex: 5;
  -ms-flex: 5;
  flex: 5;
  font-size: 1rem;
}

.c1 {
  background-color: #ffffff;
  color: #000000;
  cursor: default;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1:hover {
  background-color: #c0c0c0 !important;
}

.c1:hover .ui-item-button {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  opacity: 1.0;
}

.c0 > .ui-item,
.c0 > .ui-item-button {
  color: #ffffff;
  background-color: #004358;
}

.c0 > .ui-item:hover {
  color: #ffe11a;
  background-color: #004358 !important;
}

<AccordionItem
  className=""
  contentEditable={false}
  direction="right"
  disabled={false}
  focus={false}
  height=""
  hiddenLeftButton={false}
  hiddenRightButton={false}
  id=""
  initialToggle={false}
  layout={5}
  leftButton={null}
  location="none"
  nocollapse={true}
  noedit={false}
  nohover={false}
  noripple={false}
  obj="AccordionItem"
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onDoubleClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onMouseOut={[Function]}
  onUpdate={[Function]}
  rightButton={null}
  selected={false}
  sizing="normal"
  stacked={false}
  style={Object {}}
  testing={true}
  theme={null}
  title="Test Title"
  tooltip=""
  useedit={false}
  visible={true}
  widget={null}
  width=""
  xcss={null}
>
  <ThemeProvider
    theme={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#c0c0c0",
        "color": "#000000",
        "headerBackgroundColor": "#004358",
        "headerForegroundColor": "#ffffff",
        "headerHoverColor": "#ffe11a",
        "hoverColor": "#c0c0c0",
        "inputBorderColor": "#c0c0c0",
        "selectedBackgroundColor": "#1f8a70",
        "selectedForegroundColor": "#ffe11a",
        "tooltipBackgroundColor": "#708090",
        "tooltipForegroundColor": "#ffffff",
        "transitionDelay": "0.5s",
      }
    }
  >
    <styled.ul
      className="ui-accordionitem"
      style={Object {}}
    >
      <ul
        className="ui-accordionitem c0"
        style={Object {}}
      >
        <Item
          className=""
          contentEditable={false}
          direction="right"
          disabled={false}
          focus={false}
          height=""
          hiddenLeftButton={false}
          hiddenRightButton={false}
          id=""
          initialToggle={false}
          layout={5}
          leftButton={null}
          location="none"
          nocollapse={true}
          noedit={false}
          nohover={false}
          noripple={false}
          obj="AccordionItem"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDoubleClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onMouseOut={[Function]}
          onUpdate={[Function]}
          rightButton={null}
          selected={false}
          sizing="normal"
          stacked={false}
          style={Object {}}
          testing={true}
          theme={null}
          title="Test Title"
          tooltip=""
          useedit={false}
          visible={true}
          widget={null}
          width=""
          xcss={null}
        >
          <ThemeProvider
            theme={
              Object {
                "backgroundColor": "#ffffff",
                "borderColor": "#c0c0c0",
                "color": "#000000",
                "headerBackgroundColor": "#004358",
                "headerForegroundColor": "#ffffff",
                "headerHoverColor": "#ffe11a",
                "hoverColor": "#c0c0c0",
                "inputBorderColor": "#c0c0c0",
                "selectedBackgroundColor": "#1f8a70",
                "selectedForegroundColor": "#ffe11a",
                "tooltipBackgroundColor": "#708090",
                "tooltipForegroundColor": "#ffffff",
                "transitionDelay": "0.5s",
              }
            }
          >
            <styled.li
              className="ui-item"
              id=""
              nohover={false}
              onBlur={[Function]}
              onDoubleClick={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onMouseOut={[Function]}
              selected={false}
              style={Object {}}
            >
              <li
                className="ui-item c1"
                id=""
                onBlur={[Function]}
                onDoubleClick={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onMouseOut={[Function]}
                selected={false}
                style={Object {}}
              >
                <Title
                  className=""
                  contentEditable={false}
                  direction="right"
                  disabled={false}
                  focus={false}
                  height=""
                  hiddenLeftButton={false}
                  hiddenRightButton={false}
                  id=""
                  initialToggle={false}
                  layout={5}
                  leftButton={null}
                  location="none"
                  nocollapse={true}
                  noedit={false}
                  nohover={false}
                  noripple={false}
                  obj="AccordionItem"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onDoubleClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onMouseOut={[Function]}
                  onUpdate={[Function]}
                  rightButton={null}
                  selected={false}
                  sizing="normal"
                  stacked={false}
                  style={
                    Object {
                      "padding": "2px 4px",
                    }
                  }
                  testing={true}
                  theme={null}
                  title="Test Title"
                  tooltip=""
                  useedit={false}
                  visible={true}
                  widget={null}
                  width=""
                  xcss={null}
                >
                  <ThemeProvider
                    theme={
                      Object {
                        "backgroundColor": "#ffffff",
                        "borderColor": "#c0c0c0",
                        "color": "#000000",
                        "headerBackgroundColor": "#004358",
                        "headerForegroundColor": "#ffffff",
                        "headerHoverColor": "#ffe11a",
                        "hoverColor": "#c0c0c0",
                        "inputBorderColor": "#c0c0c0",
                        "selectedBackgroundColor": "#1f8a70",
                        "selectedForegroundColor": "#ffe11a",
                        "tooltipBackgroundColor": "#708090",
                        "tooltipForegroundColor": "#ffffff",
                        "transitionDelay": "0.5s",
                      }
                    }
                  >
                    <styled.div
                      className="ui-title-bar ripple"
                      disabled={false}
                      layout={5}
                      style={
                        Object {
                          "padding": "2px 4px",
                        }
                      }
                      visible={true}
                    >
                      <div
                        className="ui-title-bar ripple c2"
                        disabled={false}
                        style={
                          Object {
                            "padding": "2px 4px",
                          }
                        }
                      >
                        <Styled(Label)
                          className="ui-title"
                          contentEditable={false}
                          direction="right"
                          disabled={false}
                          focus={false}
                          height=""
                          hiddenLeftButton={false}
                          hiddenRightButton={false}
                          id=""
                          initialToggle={false}
                          layout={5}
                          leftButton={null}
                          location="none"
                          nocollapse={true}
                          noedit={false}
                          nohover={false}
                          noripple={false}
                          obj="AccordionItem"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={[Function]}
                          onDoubleClick={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyPress={[Function]}
                          onMouseOut={[Function]}
                          onUpdate={[Function]}
                          rightButton={null}
                          selected={false}
                          sizing="normal"
                          stacked={false}
                          style={
                            Object {
                              "padding": "2px 4px",
                            }
                          }
                          testing={true}
                          text="Test Title"
                          theme={null}
                          title="Test Title"
                          tooltip=""
                          useedit={false}
                          visible={true}
                          widget={null}
                          width=""
                          xcss={
                            Array [
                              "
	text-align: left;
	flex: 5;
",
                            ]
                          }
                        >
                          <Label
                            className="ui-title c3"
                            contentEditable={false}
                            direction="right"
                            disabled={false}
                            focus={false}
                            height=""
                            hiddenLeftButton={false}
                            hiddenRightButton={false}
                            id=""
                            initialToggle={false}
                            layout={5}
                            leftButton={null}
                            location="none"
                            nocollapse={true}
                            noedit={false}
                            nohover={false}
                            noripple={false}
                            obj="AccordionItem"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onClick={[Function]}
                            onDoubleClick={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyPress={[Function]}
                            onMouseOut={[Function]}
                            onUpdate={[Function]}
                            rightButton={null}
                            selected={false}
                            sizing="normal"
                            stacked={false}
                            style={
                              Object {
                                "padding": "2px 4px",
                              }
                            }
                            testing={true}
                            text="Test Title"
                            theme={null}
                            title="Test Title"
                            tooltip=""
                            useedit={false}
                            visible={true}
                            widget={null}
                            width=""
                            xcss={
                              Array [
                                "
	text-align: left;
	flex: 5;
",
                              ]
                            }
                          >
                            <ThemeProvider
                              theme={
                                Object {
                                  "backgroundColor": "#ffffff",
                                  "borderColor": "#c0c0c0",
                                  "color": "#000000",
                                  "headerBackgroundColor": "#004358",
                                  "headerForegroundColor": "#ffffff",
                                  "headerHoverColor": "#ffe11a",
                                  "hoverColor": "#c0c0c0",
                                  "inputBorderColor": "#c0c0c0",
                                  "selectedBackgroundColor": "#1f8a70",
                                  "selectedForegroundColor": "#ffe11a",
                                  "tooltipBackgroundColor": "#708090",
                                  "tooltipForegroundColor": "#ffffff",
                                  "transitionDelay": "0.5s",
                                }
                              }
                            >
                              <styled.span
                                className="ui-label ui-title c3"
                                contentEditable={false}
                                disabled={false}
                                innerRef={[Function]}
                                onBlur={[Function]}
                                onClick={[Function]}
                                onDoubleClick={[Function]}
                                onKeyDown={[Function]}
                                onKeyPress={[Function]}
                                sizing="normal"
                                style={
                                  Object {
                                    "padding": "2px 4px",
                                  }
                                }
                                suppressContentEditableWarning={true}
                                visible={true}
                              >
                                <span
                                  className="ui-label ui-title c3 c4"
                                  contentEditable={false}
                                  disabled={false}
                                  onBlur={[Function]}
                                  onClick={[Function]}
                                  onDoubleClick={[Function]}
                                  onKeyDown={[Function]}
                                  onKeyPress={[Function]}
                                  style={
                                    Object {
                                      "padding": "2px 4px",
                                    }
                                  }
                                  suppressContentEditableWarning={true}
                                >
                                  Test Title
                                </span>
                              </styled.span>
                            </ThemeProvider>
                          </Label>
                        </Styled(Label)>
                      </div>
                    </styled.div>
                  </ThemeProvider>
                </Title>
              </li>
            </styled.li>
          </ThemeProvider>
        </Item>
        <div
          className="ui-accordion-content"
        >
          <div>
            child content
          </div>
        </div>
      </ul>
    </styled.ul>
  </ThemeProvider>
</AccordionItem>
`;

exports[`Test making the AccordionItem invisible 1`] = `
<ThemeProvider
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "color": "#000000",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.5s",
    }
  }
>
  <styled.ul
    className="ui-accordionitem"
    style={Object {}}
  >
    <Item
      className=""
      contentEditable={false}
      direction="right"
      disabled={false}
      focus={false}
      height=""
      hiddenLeftButton={false}
      hiddenRightButton={false}
      id=""
      initialToggle={false}
      layout={5}
      leftButton={null}
      location="none"
      nocollapse={false}
      noedit={false}
      nohover={false}
      noripple={false}
      obj="AccordionItem"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onDoubleClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onMouseOut={[Function]}
      onUpdate={[Function]}
      rightButton={null}
      selected={false}
      sizing="normal"
      stacked={false}
      style={Object {}}
      testing={true}
      theme={null}
      title="Test Title"
      tooltip=""
      useedit={false}
      visible={false}
      widget={null}
      width=""
      xcss={null}
    />
  </styled.ul>
</ThemeProvider>
`;

exports[`Test retrieval of AccordionItem props object 1`] = `
Object {
  "className": "",
  "contentEditable": false,
  "direction": "right",
  "disabled": false,
  "focus": false,
  "height": "",
  "hiddenLeftButton": false,
  "hiddenRightButton": false,
  "id": "",
  "initialToggle": false,
  "layout": 5,
  "leftButton": null,
  "location": "none",
  "nocollapse": false,
  "noedit": false,
  "nohover": false,
  "noripple": false,
  "obj": "AccordionItem",
  "onBlur": [Function],
  "onChange": [Function],
  "onClick": [Function],
  "onDoubleClick": [Function],
  "onFocus": [Function],
  "onKeyDown": [Function],
  "onKeyPress": [Function],
  "onMouseOut": [Function],
  "onUpdate": [Function],
  "rightButton": null,
  "selected": false,
  "sizing": "normal",
  "stacked": false,
  "style": Object {},
  "testing": true,
  "theme": null,
  "title": "​",
  "tooltip": "",
  "useedit": false,
  "visible": true,
  "widget": null,
  "width": "",
  "xcss": null,
}
`;

exports[`Test the creation of a AccordionItem control 1`] = `
<ThemeProvider
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "color": "#000000",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.5s",
    }
  }
>
  <styled.ul
    className="ui-accordionitem test-class"
    style={Object {}}
  >
    <Item
      className="test-class"
      contentEditable={false}
      direction="right"
      disabled={false}
      focus={false}
      height=""
      hiddenLeftButton={false}
      hiddenRightButton={false}
      id=""
      initialToggle={false}
      layout={5}
      leftButton={null}
      location="none"
      nocollapse={false}
      noedit={false}
      nohover={false}
      noripple={false}
      obj="AccordionItem"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onDoubleClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onMouseOut={[Function]}
      onUpdate={[Function]}
      rightButton={null}
      selected={false}
      sizing="normal"
      stacked={false}
      style={Object {}}
      testing={true}
      theme={null}
      title="Test Title"
      tooltip=""
      useedit={false}
      visible={true}
      widget={null}
      width=""
      xcss={null}
    />
  </styled.ul>
</ThemeProvider>
`;

exports[`Test the disabling of the AccordionItem 1`] = `
<ThemeProvider
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "color": "#000000",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.5s",
    }
  }
>
  <styled.ul
    className="ui-accordionitem"
    style={Object {}}
  >
    <Item
      className=""
      contentEditable={false}
      direction="right"
      disabled={true}
      focus={false}
      height=""
      hiddenLeftButton={false}
      hiddenRightButton={false}
      id=""
      initialToggle={false}
      layout={5}
      leftButton={null}
      location="none"
      nocollapse={false}
      noedit={false}
      nohover={false}
      noripple={false}
      obj="AccordionItem"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onDoubleClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onMouseOut={[Function]}
      onUpdate={[Function]}
      rightButton={null}
      selected={false}
      sizing="normal"
      stacked={false}
      style={Object {}}
      testing={true}
      theme={null}
      title="Test Title"
      tooltip=""
      useedit={false}
      visible={true}
      widget={null}
      width=""
      xcss={null}
    />
  </styled.ul>
</ThemeProvider>
`;
